lib/spack/spack/user_environment.py

```diff
@@ -77,7 +77,8 @@ def environment_modifications_for_spec(spec, view=None, set_package_py_globals=T
     """
     spec = spec.copy()
     if view and not spec.external:
-        spec.prefix = prefix.Prefix(view.get_projection_for_spec(spec))
+        for node in spec.traverse():
+            node.prefix = prefix.Prefix(view.get_projection_for_spec(node))
```

Comment:
this is wrong, move the external check inside the loop

Reply:
also, `spec.copy()` is done at the wrong place, since the `environmen_modifications_for_spec` is called on every spec in the env, so it's creating a quadratic number of copies (ok, i'm fixing that elsewhere)